pkg/controllers/nodeclass/validation.go

```diff
@@ -324,19 +339,98 @@ func mockLaunchTemplateConfig() []ec2types.FleetLaunchTemplateConfigRequest {
 	}
 }
 
-func (v *Validation) mockOptions(ctx context.Context, nodeClaim *karpv1.NodeClaim, nodeClass *v1.EC2NodeClass, tags map[string]string) ([]*amifamily.LaunchTemplate, error) {
-	amiOptions, err := v.launchTemplateProvider.CreateAMIOptions(ctx, nodeClass, lo.Assign(nodeClaim.Labels, map[string]string{karpv1.CapacityTypeLabelKey: karpv1.CapacityTypeOnDemand}), tags)
+func (v *Validation) mockLaunchTemplateOptions(ctx context.Context, nodeClaim *karpv1.NodeClaim, nodeClass *v1.EC2NodeClass, tags map[string]string) (*amifamily.LaunchTemplate, error) {
+	amiOptions, err := v.launchTemplateProvider.CreateAMIOptions(
+		ctx,
+		nodeClass,
+		lo.Assign(nodeClaim.Labels, map[string]string{karpv1.CapacityTypeLabelKey: karpv1.CapacityTypeOnDemand}),
+		tags,
+	)
 	if err != nil {
 		return nil, err
 	}
-	return v.amiResolver.Resolve(nodeClass, nodeClaim, []*cloudprovider.InstanceType{
-		{
-			Name:         "m5.large",
-			Requirements: scheduling.NewRequirements(scheduling.NewRequirement(corev1.LabelArchStable, corev1.NodeSelectorOpIn, karpv1.ArchitectureAmd64)),
-		},
-		{
-			Name:         "m6g.large",
-			Requirements: scheduling.NewRequirements(scheduling.NewRequirement(corev1.LabelArchStable, corev1.NodeSelectorOpIn, karpv1.ArchitectureArm64)),
+
+	// Select an instance type to use for validation. If NodePools exist for this NodeClass, we'll use an instance type
+	// selected by one of those NodePools. We should also prioritize an InstanceType which will launch with a non-GPU AMI,
+	// since GPU AMIs may have a larger snapshot size than that supported by the NodeClass' blockDeviceMappings.
+	// Issue: https://github.com/aws/karpenter-provider-aws/issues/7928
+	amiMap := amifamily.MapToInstanceTypes(v.getInstanceTypesForNodeClass(ctx, nodeClass), nodeClass.Status.AMIs)
+	var instanceTypes []*cloudprovider.InstanceType
+	for _, ami := range nodeClass.Status.AMIs {
```

Comment:
Just a nit, but adding a comment can save the reader time to understand what our intention is.
```go
// First search for VariantStandard compliant non-GPU AMI.
for _, ami := range nodeClass.Status.AMIs {
```